File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Organisms/OverflowMenu/State/OverflowMenuState.swift
public enum OverflowMenuState {
    case `default`
    case expanded

    public var isExpanded: Bool {
        switch self {
        case .default: false
        case .expanded: true
        }
    }

    public var colorRole: IconColorRole {
        switch self {
        case .default: .onScrim
        case .expanded: .primary
        }
    }
}

File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/organisms/overflowmenu/state/OverflowMenuState.kt
package com.perrystreet.woof.designsystem.atomic.organisms.overflowmenu.state

import com.perrystreet.woof.designsystem.atomic.atoms.icon.roles.IconColorRole

enum class OverflowMenuState {
    Default {
        override val isExpanded = false
        override val colorRole = IconColorRole.OnScrim
    },
    Expanded {
        override val isExpanded = true
        override val colorRole = IconColorRole.Primary
    },
    ;

    abstract val isExpanded: Boolean
    abstract val colorRole: IconColorRole
}
